iOS/DuckDuckGo/AppConfiguration/AppConfiguration.swift

```diff
@@ -90,12 +92,53 @@ struct AppConfiguration {
             return
         }
 
-        do {
-            try FileManager.default.createDirectory(at: url, withIntermediateDirectories: true, attributes: nil)
-            Logger.general.info("📁 Recreated temp directory at: \(url.path)")
-        } catch {
-            Logger.general.error("❌ Failed to recreate tmp dir: \(error.localizedDescription)")
-            Pixel.fire(pixel: .failedToRecreateTmpDir, error: error)
+        let maxAttempts = 5
+        let retryInterval: TimeInterval = 1.0
+        
+        for attempt in 0..<maxAttempts {
+            do {
+                try FileManager.default.createDirectory(at: url, withIntermediateDirectories: true, attributes: nil)
+                Logger.general.info("📁 Recreated temp directory at: \(url.path)")
+                
+                if attempt > 0 {
+                    Pixel.fire(pixel: .recreateTmpSuccessOnRetry(attempt: attempt))
+                }
+                return
+            } catch {
+                Logger.general.error("❌ Failed to recreate tmp dir (attempt \(attempt)): \(error.localizedDescription)")
+                Pixel.fire(pixel: .recreateTmpAttemptFailed(attempt: attempt), error: error)
+
+                let isLastAttempt = attempt == maxAttempts - 1
+                if isLastAttempt {
+                    attemptWebViewTempDirectoryFallback(at: url)
+                    return
+                } else {
+                    Thread.sleep(forTimeInterval: retryInterval)
+                }
+            }
+        }
+    }
+    
+    private func attemptWebViewTempDirectoryFallback(at url: URL) {
+        Logger.general.info("🌐 Attempting WKWebView fallback for temp directory recreation")
+        // Create a minimal WKWebView and load content to trigger temp directory creation
+        // WebKit may have elevated privileges that could help with directory creation
+        let webView = WKWebView(frame: .zero)
+        
+        // Load about:blank to trigger WebKit's temp directory creation
+        if let aboutBlankURL = URL(string: "about:blank") {
+            let request = URLRequest(url: aboutBlankURL)
+            webView.load(request)
+        }
+        
+        let fallbackSucceeded = FileManager.default.fileExists(atPath: url.path)
```

Comment:
Ideally we should be waiting for navigation to finish to ensure comms between our and WebKit processes did happen, but maybe simple sleep for now would suffice, as we are in the investigation mode?

Reply:
I tested it and seems that simply creating WKWebview immediately recreates the directory. I would leave like this for now.